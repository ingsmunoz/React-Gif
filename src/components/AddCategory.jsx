import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = (/*{setCategories}*/ {onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }/*event*/) => {
        setInputValue(target.value);
    }

    const onSubmit = ( event )=>{
        event.preventDefault();
        // En caso que sea menor a 1 no se va insertar nada. (No pasa a la siguiente instruccion)
        if ( inputValue.trim().length <= 1) return;
        // setCategories( categories => [ inputValue, ...categories])
        
        // Borramos el contenido de nuestra caja.
        setInputValue('');
        // Lo que hace aqui es emitirlo y enviarle el valor a la funcion
        onNewCategory( inputValue.trim() );
    }

    return (

        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>

    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}