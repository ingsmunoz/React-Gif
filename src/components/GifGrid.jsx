// Primero imports de react - Terceros - Nuestros. en ese orden
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {

    // Custom Hook useFetchGifs
    const { images, isLoading } = useFetchGifs( category );

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading &&  ( <h2>Cargando...</h2> )
            }
            

            <div className="card-grid">
                {
                    // Podemos usar destructuracion o funcion que llama cada valor. o spred
                    images?.map((image) => (
                        <GifItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>


        </>
    )
}
