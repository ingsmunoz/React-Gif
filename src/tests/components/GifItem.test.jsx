import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe('Pruebas en <GifItem/>', () => {

    const title = 'Saitama';
    const url = 'https://media4.giphy.com/media/DBv3fLQGTBmJW/giphy.gif?cid=56ae463f4pe40f6st6j85hphqoqfmnb5bpq4xylrzla48ct7&ep=v1_gifs_search&rid=giphy.gif&ct=g';

    test('debe de hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe de mostrar la imagen con el URL y el ALT indicado ', () => {
        render(<GifItem title={title} url={url} />);
        // expect( screen.getByRole('img').alt ).toBe( title );

        // Mucho mejor destructuracion.
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe de mostar el titulo del componente', () => {
        render(<GifItem title={title} url={url} />);
        expect( screen.getByText( title ) ).toBeTruthy();
    })
})