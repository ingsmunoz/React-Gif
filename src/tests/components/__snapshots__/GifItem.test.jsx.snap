// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media4.giphy.com/media/DBv3fLQGTBmJW/giphy.gif?cid=56ae463f4pe40f6st6j85hphqoqfmnb5bpq4xylrzla48ct7&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
